### PR TITLE
Network Server utilities

### PR DESCRIPTION
**Pull request by rvolosatovs**
#### Summary

Various Network Server utilities extracted from https://github.com/TheThingsNetwork/lorawan-stack/pull/3052 for easier review

#### Changes

- Extract various helpers, which are used in https://github.com/TheThingsNetwork/lorawan-stack/pull/3052
- Fix allocations in LoRaWAN codec (it's a 1-line change, which does not deserve a separate PR)

#### Testing

unnecesary

#### Checklist

- [x] Scope: The referenced issue is addressed, there are no unrelated changes.
- [x] Compatibility: The changes are backwards compatible with existing API, storage, configuration and CLI, according to the compatibility commitments in `README.md` for the chosen target branch.
- [x] Documentation: Relevant documentation is added or updated.
- [x] Changelog: Significant features, behavior changes, deprecations and fixes are added to `CHANGELOG.md`.
- [x] Commits: Commit messages follow guidelines in `CONTRIBUTING.md`, there are no fixup commits left.

### CODE REVIEW

**Comment by rvolosatovs** [comment]
Blocking https://github.com/TheThingsNetwork/lorawan-stack/pull/3052

**Comment by rvolosatovs** [comment]
Setting `prio/high` as dependant issue is `prio/high`

**Review by adriansmares — APPROVED** [review]
LGTM

File: pkg/networkserver/grpc_gsns.go [hunk 1]
```diff
@@ -173,6 +173,41 @@ type contextualEndDevice struct {
...
+	}
+	switch cfList.Type {
+	case ttnpb.CFListType_FREQUENCIES:
+		for _, freq := range cfList.Freq {
+			if freq == 0 {
+				break
+			}
+			chs = append(chs, &ttnpb.MACParameters_Channel{
+				UplinkFrequency:   uint64(freq * 100),
+				DownlinkFrequency: uint64(freq * 100),
```
Comment (johanstokking) [inline pkg/networkserver/grpc_gsns.go]: Use the band's frequency multiplier (it's 200 for 2.4 GHz)

**Review by johanstokking — APPROVED** [review]

**Pull request merged**